Add design details and principles

kairoaraujo opened the pull request:
It adds to RSTUF development documentation

- More details about the RSTUF design
- Include the RSTUF design principles

kairoaraujo commented:
Review tip: Use the rendered version -> https://repository-service-tuf.readthedocs.io/en/design_details_principles/devel/design.html#

MVrachev commented on docs/source/devel/design.rst:
```diff
@@ -1,19 +1,214 @@
+
+   * RSTUF is process agnostic.
+
+     - Adding/removing artifacts goes with the existing releasing/publishing
+       process.
```
```suggestion
     - The addition or removal of artifacts doesn't interfere 
     with the existing release/publish processes.
```

MVrachev commented on docs/source/devel/design.rst:
```diff
@@ -1,19 +1,214 @@
+=====================
+
+    * Every API request is a task.
+    * RSTUF centralizes all tasks in the Message :ref:`devel/design:Message Queue`.
```
```suggestion
    * RSTUF centralizes all tasks in the :ref:`devel/design:Message Queue`.
```

MVrachev commented on docs/source/devel/design.rst:
```diff
@@ -1,19 +1,214 @@
+Backend Result
+==============
+
+* It is a centralized backend result for tasks
```
Is this a service or what is it?
The description doesn't give much information.

kairoaraujo replied:
I will add more information

kairoaraujo replied:
I added more the context for Celery, included the same to message queue

MVrachev commented on docs/source/devel/design.rst:
```diff
@@ -1,19 +1,214 @@
+RSTUF Repository Settings/Configuration
+=======================================
+
+    * :ref:`guide/general/introduction:RSTUF Components` are RSTUF
+      **stateless** about the Repository Settings, relying on
+      :ref:`devel/design:RSTUF settings`.
```
That one is a little strangely formulated.
Usually when you say `stateless` you say that something is overall stateless and not just for a particular setting.
Also, it's a little unclear what is `devel/design:RSTUF settings`.

kairoaraujo replied:
I gave more context about the stateless and fixed the reference, adding the `TUF Repository Settings` as a sub-title of `RSTUF Repository Settings/Configuration`

MVrachev commented on docs/source/devel/design.rst:
```diff
@@ -1,19 +1,214 @@
+    * :ref:`devel/design:Repository Service TUF Worker` **writes** settings
+    * :ref:`devel/design:Repository Service TUF API` **reads** settings.
+
+        .. note::
+          A unique exception is when a bootstrap process fails and the API
+          **writes** it back to ``None``.
```
This note looks a little bit out of place:
<img width="387" alt="image" src="https://github.com/repository-service-tuf/repository-service-tuf/assets/16246778/db80e476-3cf6-4dfc-9aad-a81062bc0c6b">

Is there a better place for this?

kairoaraujo replied:
The place is correct. I will reformulate the sentence.

kairoaraujo replied:
I rephrased the sentence and gave the full context of the exception.

MVrachev commented on docs/source/devel/design.rst:
```diff
@@ -1,19 +1,214 @@
+            | Example: ``ROOT_NUM_KEYS``
+        * - ``<ROLE NAME>_THRESHOLD``
+          - | ``int``
+            | keys threshold
+          - | Role threshold
```
```suggestion
          - | Role key threshold
```

MVrachev commented on docs/source/devel/design.rst:
```diff
@@ -1,19 +1,214 @@
+            | Example: ``ROOT_EXPIRATION``
+        * - ``<ROLE NAME>_NUM_KEYS``
+          - | ``int``
+            | number of keys
+          - | Role number of keys
```
```suggestion
          - | Role number of keys
```

MVrachev commented on docs/source/devel/design.rst:
```diff
@@ -1,19 +1,214 @@
+            | ``sign-<task id>``: Signing process
+        * - ``<ROLE NAME>_EXPIRATION``
+          - | ``int``
+            | days
+          - | Role Metadata expiration setting
```
I don't see why you explain the values inside the `values` column when there is a specific `description` column for that.
I mean these four settings precisly:
<img width="353" alt="image" src="https://github.com/repository-service-tuf/repository-service-tuf/assets/16246778/c1e59f28-c4ee-4a48-91ea-b7ee9ecdacce">

I have suggested below how to fix them:
```suggestion
          - | Role Metadata expiration policy in days
```

MVrachev commented on docs/source/devel/design.rst:
```diff
@@ -1,19 +1,214 @@
+Target Files and Target Roles
+=============================
+
+    * The TUF top Role Targets is a role only for delegation. This role does
+      not register target files (artifacts).
```
```suggestion    
    * The TUF top-level Targets Role is only used for delegation. 
This role does not register target files (artifacts).
```

MVrachev commented on docs/source/devel/design.rst:
```diff
@@ -1,19 +1,214 @@
+==========================
+
+* Command Line Interface for the API
+* Guide users in the process
```
```suggestion
* Guide users in the processes
```

kairoaraujo commented on docs/diagrams/2_1_rstuf.puml:
```diff
@@ -45,17 +40,21 @@ System_Boundary(rstuf, "Repository Service for TUF ") {
+ContainerDb(storage_service, "Storage Service", "Local/SaaS Storage", $tags="storage_service") #LightGrey
+ContainerDb(sql, "SQL Server", "SQL Server", $tags="sql_db") #SlateGray
 
 Rel(rstuf_cli, rstuf_rest_api, "REPOSITORY-SERVICE-TUF-CLI", "HTTP")
```
I added this as HTTP(S)

kairoaraujo commented on docs/diagrams/2_1_rstuf.puml:
```diff
@@ -45,17 +40,21 @@ System_Boundary(rstuf, "Repository Service for TUF ") {
 
 Rel(rstuf_cli, rstuf_rest_api, "REPOSITORY-SERVICE-TUF-CLI", "HTTP")
 Rel_U(queue, rstuf_repo_worker, "Consumer", "Tasks")
+Rel_U(redis, rstuf_repo_worker, "Read/Write", "Settings")
```
I did it and fixed other relationships

kairoaraujo commented on docs/source/devel/design.rst:
```diff
@@ -1,19 +1,214 @@
-- `repository-service-tuf-api <https://repository-service-tuf.readthedocs.io/projects/rstuf-api/>`_
-- `repository-service-tuf-worker <https://repository-service-tuf.readthedocs.io/projects/rstuf-worker/>`_
-- `repository-service-tuf-cli <https://repository-service-tuf.readthedocs.io/projects/rstuf-cli/>`_
+     - RSTUF is language agnostic, making any language use/integrate easily.
```
I rephrased it, using part of your suggestion but tried to clarify all the context.

kairoaraujo commented:
I added the suggestions and added some improvements
[New rendered version is available here](https://repository-service-tuf.readthedocs.io/en/design_details_principles/devel/design.html#)

MVrachev commented on docs/source/devel/design.rst:
```diff
@@ -1,19 +1,229 @@
+      :ref:`devel/design:TUF Repository Settings` using
+      `Dynaconf <https://www.dynaconf.com>`_.
+
+    * :ref:`devel/design:Repository Service TUF Worker` **writes** settings
```
```suggestion
    * :ref:`devel/design:Repository Service TUF Worker` **writes** settings.
```

MVrachev commented on docs/source/devel/design.rst:
```diff
@@ -1,19 +1,229 @@
+Redis
+=====
+
+* It is a centralized key/cache service
```
```suggestion
* It is a centralized key/cache service.
```

MVrachev commented on docs/source/devel/design.rst:
```diff
@@ -1,19 +1,229 @@
+* Stores :ref:`devel/design:RSTUF Repository Settings/Configuration`
+* Optional:
+
+  - Used as :ref:`devel/design:Message Queue`
+  - Used as :ref:`devel/design:Backend Result`
```
```suggestion
  - Used as :ref:`devel/design:Message Queue`.
  - Used as :ref:`devel/design:Backend Result`.
```

MVrachev reviewed (changes requested):
I added some small comments for consistency.
It's better than the last time.
Good document!

kairoaraujo commented on docs/source/devel/design.rst:
```diff
@@ -1,19 +1,229 @@
+
+   * RSTUF is process agnostic.
+
+     - Add/Remove artifacts doesn't interfere with the existing organization
+       release/publish processes.
```
It is still strange for me, maybe:
`Add/Remove artifacts to TUF Metadata doesn't interfere with the existing release/publish artifact processes.`

kairoaraujo commented on docs/source/devel/design.rst:
```diff
@@ -1,19 +1,229 @@
+TUF Repository Setting are key configuration for the Metadata Repository
+operations.
+
+    .. list-table:: RSTUF reserved settings/configuration
```
I didn't mention it because there is no use case for this. 
It can be removed in the future.
